src/main.js

```diff
@@ -9,42 +9,59 @@ import { showError, showSuccess } from '@nextcloud/dialogs'
 
 	const FilesPlugin = {
 		attach(fileList) {
+			const displayName = t('files_zip', 'Compress to Zip')
+
+			const actionHandler = (files) => {
+				const sum = files.reduce((carry, file) => file.size + carry, 0)
+				if (MAX_COMPRESS_SIZE !== -1 && sum > MAX_COMPRESS_SIZE) {
+					showError(t('files_zip', 'Only files up to {maxSize} can be compressed.', {
+						maxSize: formatFileSize(MAX_COMPRESS_SIZE),
+					}))
+					return
+				}
+
+				const parentFolderName = files.length === 1 ? (files[0].name ?? files[0].get('name')) : fileList.getCurrentDirectory().split('/').slice(-1).pop()
```

Comment:
**Question:** What the reason about this line? `(files[0].name ?? files[0].get('name'))`

Reply:
The reason here is that the single file action provides a FileInfoModel backbone.js model object while the multi select action get provided a list of objects, so we need different cases here to get the name.

https://github.com/nextcloud/server/blob/4763a1263f9831ce06fbc200e2ec70f772ba735b/apps/files/js/fileactions.js#L859
https://github.com/nextcloud/server/blob/4763a1263f9831ce06fbc200e2ec70f772ba735b/apps/files/js/fileinfomodel.js#L32

vs.

https://github.com/nextcloud/server/blob/532ea001a571078f643a70a906e15b32afebde6a/apps/files/js/filelist.js#L827
https://github.com/nextcloud/server/blob/532ea001a571078f643a70a906e15b32afebde6a/apps/files/js/filelist.js#L1390-L1401